.github/workflows/phpcs.yml

```diff
@@ -31,59 +35,134 @@ jobs:
         id: get-composer-cache-dir
         run: |
           echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
-      - uses: actions/cache@v2
-        id: composer-cache
+      - name: Cache Composer dependencies
+        uses: actions/cache@v2
         with:
           path: ${{ steps.get-composer-cache-dir.outputs.dir }}
           key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
           restore-keys: |
             ${{ runner.os }}-composer-
 
-      - uses: "shivammathur/setup-php@v2"
-        with:
-          php-version: ${{ inputs.php_version }}
-      - uses: "ramsey/composer-install@v2"
-
-      - name: "Give permissions"
+      # ------------------------------------------------------------------------------
+      # Check for PHPCS
+      # ------------------------------------------------------------------------------
+      - name: Check PHPCS installation
         run: |
-          sudo chown -R root:root $GITHUB_WORKSPACE
+          if [ ! -f vendor/bin/phpcs ]; then
+            echo "PHPCS not found in vendor/bin, installing..."
+            composer require "squizlabs/php_codesniffer=*"
+          else
+            echo "PHPCS found in vendor/bin"
+          fi
+          vendor/bin/phpcs --version
```

Comment:
i think i prefer how it was - meaning setup php action with composer instead of installing any version of phpcs. It is a bit slower, but can evolve along with each repo's composer.json and it wouldn't cause inconsistencies local vs GH action